internal/install/application_configuration.go

```diff
@@ -83,7 +89,9 @@ func selectElasticAgentImageName(version string) string {
 	v, err := semver.NewVersion(version)
 	if err != nil {
 		logger.Errorf("stack version not in semver format (value: %s): %v", v, err)
-	} else if !v.LessThan(elasticAgentCompleteFirstSupportedVersion) {
+	} else if !v.LessThan(elasticAgentCompleteFirstSupportedVersion) && v.LessThan(elasticAgentCompleteOwnNamespaceVersion) {
+		return elasticAgentCompleteLegacyImageName
+	} else if !v.LessThan(elasticAgentCompleteOwnNamespaceVersion) {
 		return elasticAgentCompleteImageName
```

Comment:
Nit. Simplify these ifs:
```
	if err != nil {
		logger.Errorf("stack version not in semver format (value: %s): %v", v, err)
		return elasticAgentImageName
	}
	if !v.LessThan(elasticAgentCompleteOwnNamespaceVersion) {
		return elasticAgentCompleteImageName
	}
	if !v.LessThan(elasticAgentCompleteFirstSupportedVersion) {
		return elasticAgentCompleteLegacyImageName
	}
	return elasticAgentImageName
```

This would make more explicit what image we return in case of error when parsing the version. I wonder why we don't return the newest one (`elasticAgentCompleteImageName`).

Reply:
Fixed.

> This would make more explicit what image we return in case of error when parsing the version. I wonder why we don't return the newest one (elasticAgentCompleteImageName).

It was implemented this way to ensure it's working also for the older stack versions, when the "complete" image wasn't available.